### PR TITLE
Make changevein affect connected veins

### LINKED ISSUE #4519: changevein should follow veins across map blocks
instead of stopping at the edge of the current map block

### PR DESCRIPTION
**Pull request by tatoyoda600**
Fixes #4519
changevein will now convert any connected veins in the same tile and neighboring tiles.

### CODE REVIEW

File: plugins/changevein.cpp [hunk 1]
```diff
@@ -24,6 +24,188 @@ DFHACK_PLUGIN("changevein");
...
+            continue;
+        if (evt->getType() != block_square_event_type::mineral)
+            continue;
+
+        df::block_square_event_mineralst* cur = (df::block_square_event_mineralst*)evt;
+        if (cur->inorganic_mat == mask.mat)
+        {
+            for (uint8_t j = 0; j < 15; j++)
+            {
+                uint16_t bitmask = cur->tile_bitmask.bits[j] | cur->tile_bitmask.bits[std::max(j - 1, 0)] | cur->tile_bitmask.bits[std::min(j + 1, 15)];
```
Comment (myk002) [inline plugins/changevein.cpp:116]: isn't j-1 == 255 since it's a uint8_t and has a starting value of 0?

File: plugins/changevein.cpp [hunk 2]
```diff
@@ -24,6 +24,188 @@ DFHACK_PLUGIN("changevein");
...
+
+        df::coord pos = df::coord(block->map_pos.x + offset.x, block->map_pos.y + offset.y, block->map_pos.z);
+        df::map_block* newBlock = Maps::getTileBlock(pos);
+
+        if (!newBlock)
+            continue;
+
+        for (size_t j = 0; j < newBlock->block_events.size(); j++)
+        {
+            df::block_square_event* evt = newBlock->block_events[j];
```
Comment (myk002) [inline plugins/changevein.cpp:155]:
```suggestion
        for (auto evt : newBlock->block_events)
        {
```

Comment (myk002) [reply plugins/changevein.cpp:155]: same for other similar for loops in this file. the original code was written before this syntax was available in C++, but there's no reason not to use the modern syntax

**Review by myk002 — COMMENTED** [review]
nice work!

please add a line to the changelog under "Misc Improvements" for the "Future" version

also, please document the change in behavior in `docs/changevein.rst`

File: plugins/changevein.cpp (hunk 1 above)
Comment (tatoyoda600) [reply plugins/changevein.cpp]: Great catch! Got used to using uint as if it was an int and didn't notice.

**Review by myk002 — APPROVED** [review]

**Pull request merged**